crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1562,6 +1651,18 @@ unsafe impl<T: SystemParam> SystemParam for ParamSet<'_, '_, Vec<T>> {
         Vec::new()
     }
 
+    #[inline]
+    unsafe fn validate_param(
+        state: &Self::State,
+        system_meta: &SystemMeta,
+        world: UnsafeWorldCell,
+    ) -> bool {
+        state
+            .iter()
+            .all(|state| T::validate_param(state, system_meta, world))
+    }
```

Comment:
Shouldn't this return `true` since `ParamSet::get_param` cannot panic? Or does this want to prevent panics when calling `.p0()`, `.p1()` etc etc? Though the user might not call them when they would panic, so this would silently change the behaviour in those cases.

Reply:
If any parts of the set are invalid, the system won't run.
Panicking per variant is something to be considered when systems can return `Result` and we can use `?`.

Reply:
We can remove validation from param set and keep the panicking API for now
@alice-i-cecile?

Reply:
Let's cut it from this PR and keep the panicking API. `ParamSet` is niche and that's easy follow-up.

Reply:
I'll consider this a yes, the change itself is trivial